### ISSUE TITLE
Looped effects don't cleanup after each-other

### ISSUE THREAD

**Issue by WebReflection** [preactjs/signals#152]
I am not sure if this is related to other bugs but I'll leave here [another test](https://github.com/WebReflection/usignal/blob/main/test/test.js#L278-L300) I've added to [usignal](https://github.com/WebReflection/usignal#readme) to code cover that case too:

```js
  const {assert} = console;
  (function loopedEffects() {
    const invokes = [];
    const num = signal(0);
    let loop = 2;

    effect(() => {
      invokes.push(num.value);
      for (let i = 0; i < loop; i++)
        effect(() => {
          invokes.push(num.value + i);
        });
    });

    assert(invokes.length === 3, 'looped effects not working');
    assert(invokes.join(',') === '0,0,1', 'looped values not matching');
  
    invokes.splice(0);
    loop = 1;
    num.value = 1;

    // this is actually 4 in preact, usignal and solid-js work just fine here
    assert(invokes.length === 2, 'looped effects not working after changes');
    assert(invokes.join(',') === '1,1', 'looped values not matching after changes');
  })();
```

The logic of the test is that shrinking loops should not side-effect previous loops not part of the outer effect anymore.

Feel free to ask more, if needed 👋

**Comment by jviide** [comment]
I'd like to point out that _nested_ effects are supported, but automatically disposing nested effects is not.

```ts
```

**Comment by WebReflection** [comment]
> nested effects are supported, but automatically disposing nested effects is not.

can we say then that nested effects are not very well supported? thw bigger picture is having components as array where these can be more or less (a TODO list, as example) and those dropped might side-effect unexpectedly if not disposed on "unmount" or removal. Both solid-js and usignal support this scenario, it'd be great to have it in here too 👋

**Comment by marvinhagemeister** [comment]
Fixed by #158
